### ISSUE TITLE
Some code cleanup in m2t struct

### ISSUE THREAD

**Issue by egeerardyn** [matlab2tikz/matlab2tikz#885]
Some fields of our `m2t` god struct could do with a rename for clarity/brevity or could be grouped for clarity. This is related #848 and #477. In fact, there is a lot of overlap (or duplication) with #477, but here I want to list a simple renames (or putting things in structs) such that these can happen before the next release.

Currently, the `m2t` struct looks like:
```
                   cmdOpts: [1x1 struct]
            currentHandles: [1x1 struct]
                 transform: []
           pgfplotsVersion: [1 3]
                     about: [1x1 struct]
                       tol: 1e-15
              imageAsPngNo: 0
                dataFileNo: 0
       automaticLabelIndex: 0
                colorDepth: 48
            colorPrecision: 1.5259e-05
               colorFormat: '%0.5f'
                   content: [1x1 struct]
                  preamble: '\usepackage[T1]{fontenc}…'
        extraRgbColorNames: {'mycolor1'  'mycolor2'  'mycolor3'  'mycolor4'  'mycolor5'  'mycolor6'  'mycolor7'}
        extraRgbColorSpecs: {1x7 cell}
                        ff: '%.15g'
              tikzFileName: '/private/tmp/tpbedf8b7c_6cef_4a4f_9aca_c41ef7aa4364'
          relativeDataPath: ''
                  dataPath: '/private/tmp'
             legendHandles: [1x0 GraphicsPlaceholder]
               cbarHandles: [1x0 double]
               scribeLayer: [0x0 GraphicsPlaceholder]
       relevantAxesHandles: [1x1 Axes]
           axesBoundingBox: [0 0 1 1]
            axesContainers: {[1x1 struct]}
            PlotyyLabelNum: []
             xAxisReversed: 0
             yAxisReversed: 0
             zAxisReversed: 0
    currentHandleHasLegend: 0
```

at the end of a conversion.

A few things that I think are easy:
 - [x] `m2t.cmdOpts.Results` -> `m2t.args` (these are the command line arguments, we never check any other fields, except maybe in the top level)
 - [x] Create a struct `m2t.color` containing `extraRgbColorSpecs, extraRgbColorNames, colorFormat, colorDepth, colorPrecision` where we obviously drop `color` from their names
 - [x] `axesContainers` -> `axes` (or even `ax`). We use this thing a lot, and there is little added value to knowing that they contain axes instead of representing axes (in fact, they are our internal representation of an axis). This will clean up a lot of code, I guess.
 - [x] `currentHandles` (contains `gca`, `gcf`, ...) -> `handles` and also add `legendHandles`, `cbarHandles`, `scribeLayer`, `relevantAxesHandles` (with each time `handles` removed)
 - [x] `imageAsPngNo`, `PlotyyLabelNum`, `dataFileNo`, `automaticLabelIndex` together in a struct `counter` and remove their suffix `No`/`Num`/`Index`

**Comment by PeterPablo** [comment]
Go for it! It's a pitty I currently can't really help with development.

**Comment by egeerardyn** [comment]
Closed by #883.
